Description: Check if we are running in a virtual machine before wasting time
running virtualbox guest things
Author: Kevin Puetz (puetzk)

Origin: https://bugs.launchpad.net/ubuntu/+source/virtualbox/+bug/1762491/comments/9
Last-Update: 2018-07-19

Index: virtualbox/src/VBox/Additions/x11/Installer/98vboxadd-xclient
===================================================================
--- virtualbox.orig/src/VBox/Additions/x11/Installer/98vboxadd-xclient
+++ virtualbox/src/VBox/Additions/x11/Installer/98vboxadd-xclient
@@ -25,6 +25,17 @@
 # SPDX-License-Identifier: GPL-3.0-only
 #
 
+in_virtual_machine()
+{
+	if [ -z "$(lspci -d 80ee:cafe)" ]; then
+	echo "VirtualBox Additions disabled, not in a Virtual Machine" >&2
+	return 1
+	fi
+
+	return 0
+}
+in_virtual_machine || return
+
 # Sanity check: if non-writeable PID-files are present in the user home
 # directory VBoxClient will fail to start.
 for i in $HOME/.vboxclient-*.pid; do
